src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -80,6 +80,7 @@ export class TerminalService extends Disposable implements ITerminalService {
 	private _shutdownWindowCount?: number;
 
 	private _editable: { instance: ITerminalInstance; data: IEditableData } | undefined;
+	private _instanceDisposeListeners: Map<ITerminalInstance, DisposableStore> = new Map();
```

Comment:
We want `this._register(new DisposableMap())` here so its entries get disposed when the service does.

Reply:
that takes just a disposable, not an array of them

Reply:
maybe I could create one disposable to store the array in though?